src/haxesharp/collections/Linq.hx

```diff
@@ -58,6 +59,60 @@ class Linq<T>
     }
 
     /**
+    If a predicate is supplied, returns the only element to satisfy the predicate.  Throws an exception if more than one such element exists.
+    If a predicate is not supplied, return the only element in the array.  Throws an exception if the array contains more than one element.
+    In both cases, and exception is thrown if the provided array is null.
+    */
+    public static function single<T>(array:Array<T>, ?predicate:T->Bool):T
+    {
+        if (array == null)
+        {
+            throw new ArgumentNullException('"array" argument cannot be null.');
+        }
+
+        if (predicate == null)
+        {
+            if (array.length == 1)
+            {
+                return array[0];
+            }
+            else if (array.length == 0)
+            {
+                throw new InvalidOperationException('"array" length cannot be 0.');
+            }
+            else
+            {
+                throw new InvalidOperationException('"array" length cannot be greater than 1.');                

```

Comment:
Actually, it can be one; can call `[1, 2, 3].first((n) => n == 1`. This should be removed.

Reply:
I was actually following the cases where .NET's Enumerable.Single<TSource> Method (IEnumerable<TSource>) throws exceptions, as shown here: https://msdn.microsoft.com/en-us/library/bb155325(v=vs.110).aspx#Anchor_1

But if that's not useful for us to have in haxesharp, I can remove that.  If that's the case, we probably wouldn't need the option to call `single` without a `predicate` at all, and could make that argument non-optional.

Reply:
Ah, I didn't realize this was in an `if (predicate == null)` statement. Disregard my comment please.